core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1030,8 +1034,22 @@ public String allocationId() {
         }
     }
 
+    /**
+     * The {@code ReplicasProxy} is an implementation of the {@code Replicas}
+     * interface that performs the actual {@code ReplicaRequest} on the replica
+     * shards. It also encapsulates the logic required for failing the replica
+     * if deemed necessary as well as marking it as stale when needed.
+     */
     final class ReplicasProxy implements ReplicationOperation.Replicas<ReplicaRequest> {
 
+        private final ClusterService clusterService;
+        private final TransportReplicationAction replicationAction;
+
+        ReplicasProxy(ClusterService clusterService, TransportReplicationAction replicationAction) {
```

Comment:
this is an inner class anyway, why pass these along? If you want to make it a static inner class, I'm good with it, although I'm not sure if it's worth the extra clutter.